packages/framework/src/types/discover.types.ts

```diff
@@ -65,10 +65,16 @@ export type DiscoverWorkflowOutput = {
   description?: string;
 };
 
-// eslint-disable-next-line @typescript-eslint/no-explicit-any
-export type Workflow<T_Payload = any> = {
+/**
+ * A workflow resource.
+ *
+ * @property `id` - The unique identifier for the workflow.
+ * @property `trigger` - The function to trigger the workflow.
+ * @property `discover` - The function to discover the workflow definition.
+ */
+export type Workflow<T_Payload = never> = {
```

Comment:
There is never a case where `T_Payload` is not inferred, so we make it as such. This prevents using an inferred Workflow payload without a corresponding `payloadSchema`.